client/src/components/ViewModeToggle/ViewModeToggle.js

```diff
@@ -0,0 +1,224 @@
+import React, { PropTypes, Component } from 'react';
+import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+import i18n from 'i18n';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { selectEditMode, selectPreviewMode, selectSplitMode } from 'state/viewMode/viewModeActions';
+import classNames from 'classnames';
+
+/* global window */
+class ViewModeToggle extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggle = this.toggle.bind(this);
+    this.state = {
+      dropdownOpen: false
+    };
+
+    this.handleSplitSelect = this.handleSplitSelect.bind(this);
+    this.handlePreviewSelect = this.handlePreviewSelect.bind(this);
+    this.handleEditSelect = this.handleEditSelect.bind(this);
+  }
+
+  getIconClass() {
+    const { activeState } = this.props;
+
+    switch (activeState) {
+          case 'edit':
+              return 'font-icon-edit-write';
+          case 'preview':
+              return 'font-icon-eye';
+          default:
+              return 'font-icon-columns';
```

Comment:
the indent for these lines seem off, they're 4 spaces instead of 2.
Could you have a map of the classes somewhere? possibly provided through props, or another extensible means. It would make customising the icons by an external author possible.